cmd/nodepool/kubevirt/create.go

```diff
@@ -0,0 +1,115 @@
+package kubevirt
+
+import (
+	"context"
+	"errors"
+	"os"
+	"os/signal"
+	"syscall"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/cmd/nodepool/core"
+	"github.com/spf13/cobra"
+	apiresource "k8s.io/apimachinery/pkg/api/resource"
+	kubevirtv1 "kubevirt.io/api/core/v1"
+	capikubevirt "sigs.k8s.io/cluster-api-provider-kubevirt/api/v1alpha1"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type KubevirtPlatformCreateOptions struct {
+	Memory             string
+	Cores              uint32
+	ContainerDiskImage string
+}
+
+func NewCreateCommand(coreOpts core.CreateNodePoolOptions) *cobra.Command {
+	platformOpts := KubevirtPlatformCreateOptions{
+		Memory:             "4Gi",
+		Cores:              2,
+		ContainerDiskImage: "",
+	}
+	cmd := &cobra.Command{
+		Use:          "kubevirt",
+		Short:        "Creates basic functional NodePool resources for KubeVirt platform",
+		SilenceUsage: true,
+	}
+
+	cmd.Flags().StringVar(&platformOpts.Memory, "memory", platformOpts.Memory, "The amount of memory which is visible inside the Guest OS (type BinarySI, e.g. 5Gi, 100Mi)")
+	cmd.Flags().Uint32Var(&platformOpts.Cores, "cores", platformOpts.Cores, "The number of cores inside the vmi, Must be a value greater or equal 1")
+	cmd.Flags().StringVar(&platformOpts.ContainerDiskImage, "containerdisk", platformOpts.ContainerDiskImage, "A reference to docker image with the embedded disk to be used to create the machines")
+
+	cmd.Run = func(cmd *cobra.Command, args []string) {
+		ctx, cancel := context.WithCancel(context.Background())
+		sigs := make(chan os.Signal, 1)
+		signal.Notify(sigs, syscall.SIGINT)
+		go func() {
+			<-sigs
+			cancel()
+		}()
+
+		if err := CreateNodePool(ctx, coreOpts, platformOpts); err != nil {
+			log.Error(err, "Failed to create nodepool")
+			os.Exit(1)
+		}
+	}
+
+	return cmd
+}
+
+func CreateNodePool(ctx context.Context, coreOpts core.CreateNodePoolOptions, platformOpts KubevirtPlatformCreateOptions) error {
+	return coreOpts.CreateNodePool(ctx, platformOpts)
+}
+
+func (o KubevirtPlatformCreateOptions) UpdatePlatformParams(ctx context.Context, nodePool *hyperv1.NodePool, hcluster *hyperv1.HostedCluster, client crclient.Client) error {
+	// TODO (nargaman): replace with official container image, after RFE-2501 is completed
+	// As long as there is no official container image
+	// The image must be provided by user
+	// Otherwise it must fail
+	if o.ContainerDiskImage == "" {
```

Comment:
why do we have this here instead of making the field built-in required in the cli?

Reply:
Fixed